src/screens/CardScreen.tsx

```diff
@@ -0,0 +1,339 @@
+import React, { useState } from 'react';
+import { StyleSheet, Text, View, StatusBar, Pressable } from 'react-native';
+
+import useOnPressHandlers from '../hooks/useOnPressHandlers';
+
+import StyleBase from '../styles/StyleBase';
+
+import colors from '../constants/colors';
+
+import { Benefit } from '../types';
+
+import TopBar from '../components/Bars/TopBar';
+import TapBar from '../components/Bars/TapBar';
+import CardTile from '../components/Cards/CardTile';
+import Tile from '../components/Miscellaneous/Tile';
+import BenefitList from '../components/Benefits/BenefitList';
+import CustomButton from '../components/Miscellaneous/CustomButton';
+import BoxContainer from '../components/Miscellaneous/BoxContainer';
+
+const SHOW_VIRTUAL = 1;
+
+//chaos
+
+let benefitsToAdd = [];
+
+interface CardInfoScreenProps {
+  navigation: any; //proper type
+  route: any; //proper type
+}
+
+export default function CardScreen({ navigation, route }: CardInfoScreenProps) {
+  /*
+
+  todo:
+  needs some styling
+  tested on pixel 6 pro (1440x3120), components MAY NOT fit properly on other models
+
+  proper account balance from api
+  proper descriptions
+  proper everything (after api implementation)
+  */
+
+  const { Card } = route.params;
+
+  type CardInfoScreenState = 'benefits' | 'business';
+  type ScreenState = 'card' | 'benefit' | 'claimBenefits';
+
+  const [cardInfoState, setInfo] = useState<CardInfoScreenState>('business');
+  const [screenState, setScreen] = useState<ScreenState>('card');
+  const [benefit, setBenefit] = useState<Benefit>();
+  const [accountBalance, setBalance] = useState(Card.content.points);
+
+  const { onPressBack } = useOnPressHandlers();
+
+  const AddBenefit = (benefit) => {
+    if (accountBalance < benefit.price) {
+      alert('Insufficient points'); //temp
+      return;
+    }
+    let item = benefitsToAdd.find((x) => x.id === benefit.publicId);
```

Comment:
A jakby po prostu trzymać mapę <klucz, benefit> gdzie klucz to id benefitu? nie trzeba przeszukiwać listy za każdym razem

Reply:
Do tego użyłbym lepszej nazwy zamiast `x`, ale to szczególik :P